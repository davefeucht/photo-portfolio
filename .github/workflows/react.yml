# This workflow will do a clean install of node dependencies, build the source code and run tests across different versions of node
# For more information see: https://help.github.com/actions/language-and-framework-guides/using-nodejs-with-github-actions

name: React CI/CD

on:
  push:
    branches: [ master ]
  pull_request:
    branches: [ master ]

jobs:
  build:

    runs-on: ubuntu-latest

    strategy:
      matrix:
        node-version: [18.x]

    steps:
    - name: Checkout code
      uses: actions/checkout@v2

    - name: Install Node.js ${{ matrix.node-version }}
      uses: actions/setup-node@v1
      with:
        node-version: ${{ matrix.node-version }}

    - name: Enable corepack
      run: corepack enable

    - name: Install dependencies
      run: yarn install

    - name: Build project
      run: npm run build --if-present

    - name: Run tests
      run: npm test
