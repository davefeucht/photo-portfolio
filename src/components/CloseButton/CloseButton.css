.close-button {
  display: grid;
  grid-column: 2;
  grid-template-columns: 100%;
  grid-template-rows: 100%;
  background-color: #000000;
}

.close-button-content {
  width: 30px;
  height: 30px;
  margin: auto;
}