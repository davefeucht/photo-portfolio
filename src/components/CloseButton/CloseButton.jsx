/****************
* CloseButton component displays a close button
****************/

import React, { useEffect } from "react";
import { observer } from 'mobx-react';
import './CloseButton.css';

const CloseButton = observer(({ closeFunction }) => {
  useEffect(() => {
    console.log(document.querySelector('.close-button').clientHeight);
    document.querySelector('.close-button').style.width = `${document.querySelector('.close-button').clientHeight}px`;
  })
  return(
    <div className="close-button" onClick={closeFunction.bind(this)}>X</div>
  )
});

CloseButton.displayName = 'CloseButton';

export default CloseButton;