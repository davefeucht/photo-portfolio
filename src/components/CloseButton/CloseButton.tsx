/** **************
* CloseButton component displays a close button
*************** */

import './CloseButton.css';

import * as React from 'react';
import { Link, useParams } from 'react-router-dom';

const CloseButton = () => {
    const { categoryId } = useParams();

    return (
        <div className="close-button">
            <Link to={`/category/${categoryId}`}>
                <div className="close-button-content"><img src="./assets/images/close-button.svg" width="30px" height="30px" /></div>
            </Link>
        </div>
    );
};

CloseButton.displayName = 'CloseButton';

export default CloseButton;
