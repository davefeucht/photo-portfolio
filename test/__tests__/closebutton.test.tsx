import { render } from '@testing-library/react';
import CloseButton from 'components/CloseButton/CloseButton';
import * as React from 'react';
import { MemoryRouter } from 'react-router-dom';

test('CloseButton displays', async () => {
    const { container } = render(
        <MemoryRouter>
            <CloseButton />
        </MemoryRouter>
    );
    expect(container.firstChild).toMatchSnapshot();
});
