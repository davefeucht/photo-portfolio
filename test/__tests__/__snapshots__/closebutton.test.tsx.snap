// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CloseButton displays 1`] = `
<div
  class="close-button"
  role="button"
>
  <a
    href="/category/undefined"
  >
    <div
      class="close-button-content"
    >
      <img
        height="30px"
        src="./assets/images/close-button.svg"
        width="30px"
      />
    </div>
  </a>
</div>
`;
